### PR TITLE
Simplify `StrongholdAdapterBuilder`

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This PR came about as a result of investigating #549. I found that the handling of this builder type was overly complicated and could be simplified.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: sdk/src/client/stronghold/mod.rs [hunk 1]
```diff
@@ -74,11 +74,9 @@ use super::{storage::StorageAdapter, utils::Password};
 /// A wrapper on [Stronghold].
 ///
 /// See the [module-level documentation](self) for more details.
-#[derive(Builder, Debug)]
```
Comment (thibault-martinez) [inline sdk/src/client/stronghold/mod.rs:77]: Really happy about this, I've always hated it

File: sdk/src/client/stronghold/mod.rs [hunk 2]
```diff
@@ -52,14 +52,14 @@ mod migration;
...
 mod storage;
 
+use alloc::sync::Weak;
 use std::{
     path::{Path, PathBuf},
     sync::Arc,
     time::Duration,
 };
 
-use derive_builder::Builder;
```
Comment (thibault-martinez) [inline sdk/src/client/stronghold/mod.rs:61]: Can the dependency be removed? I think it was the only use

File: sdk/src/client/stronghold/mod.rs
Comment (thibault-martinez) [inline sdk/src/client/stronghold/mod.rs:1]: There is a comment about derive_builder L167

File: sdk/src/client/stronghold/mod.rs (hunk 2 above)
Comment (DaughterOfMars) [reply sdk/src/client/stronghold/mod.rs:61]: I think it would tell me if it was unused.

Comment (DaughterOfMars) [reply sdk/src/client/stronghold/mod.rs:61]: Oh, misread

Comment (thibault-martinez) [reply sdk/src/client/stronghold/mod.rs:61]: Idk, udeps is notoriously full or misses and false positives. It really looks unused to me

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**